fitbenchmarking/fitbenchmarking
# Decouple mantid

#### Description of Work

FitBenchmarking can now be run independently of mantid when it comes to NIST problems. The file `example_runScripts.py` has now been modified to use scipy as the software to fit NIST problems. 

An additional file `example_runScripts_mantid.py` has been added to use mantid as a software to run on different type of problems (FitBenchmark problems still require modules in mantid to be analysed). In this file, there is an algorithm which checks whether mantid is installed or not. If mantid is not yet installed, there will be a message on console giving instruction of how to install mantid.

For FitBenchmark problems, their data files have been converted from the NeXuS file type to .txt type. 

A script has also been added to validate the data errors in the FitBenchmark data files. For data error values of 'nan' or 0, they would all be changed to 1.

#### Testing Instructions

1. Run `example_runScripts.py` on console and observe the results
2. Run `example_runScripts_mantid.py` on console without mantid installed
3. Run `example_runScripts_mantid.py` on console with mantid installed
4. Add a new set of formatted FitBenchmark data with 0 or nan data error values and run `benchmark_problems/format_data_file.py`
